### PR TITLE
6-pm-d: LL course form validation

### PR DESCRIPTION
**Pull request by lukelewis66**
This PR fixes #63 

Previously, admins could add empty courses to the database, with no sort of validation being done. The same issue is also happening when an admin edits a course, and leaves a field blank. What this PR aims to do is to stop this behavior. For both course add and edit, there are validations both in the frontend and the backend, which are listed below.

- All fields in the course form are now required--form won't submit until they are all filled out.

- Validation on quarter: has to be in the form [first letter of quarter][last 2 digits of year]
e.g (2020 quarters): F20 / W20 / S20 / M20

- Validation on email: has to end with "@ucsb.edu"

- Back end wont save a course if a course with the same name and quarter already exists.

- Notifications are shown using "Toast" popups using react-toast-notifications. On success a green one will appear. On failure a red one.

### CODE REVIEW

**Comment by pconrad** [comment]
@lukelewis66 you need to get this green on CI before it can be merged.  If you need help with that, please:
* ask in section today
* ask during office hours
* ask on Slack

**Comment by lukelewis66** [comment]
Quarter format needs to be changed to "YYYYQ", where Q: 1, 2, 3, 4 for winter, spring, summer, fall respectively

**Comment by pconrad** [comment]
This PR needs a code review from at least one other member of the team before the staff will look at it.

File: javascript/src/main/components/Footer/AppFooter.js
Comment (ssmith563) [inline javascript/src/main/components/Footer/AppFooter.js:6]: I don't understand why you changed the footer here. It might've been a mistake

**Review by willmori — COMMENTED** [review]
Everything looks good, just remember to change back AppFooter.js

**Review by ssmith563 — DISMISSED** [review]
Everything looked good other than the small "abcd" on the footer. Seems like it was a mistake but doesn't affect anything about the actual project

File: javascript/src/main/components/Footer/AppFooter.js
Comment (lukelewis66) [reply javascript/src/main/components/Footer/AppFooter.js]: Just changed it back with my most recent commit.

**Review by ssmith563 — APPROVED** [review]
Looks good

**Review by pconrad — APPROVED** [review]
LGTM!

**Comment by pconrad** [comment]
40 pts!

**Pull request merged**
